### PR TITLE
Move ScopeTree and BlockChain to Containers

### PR DESCRIPTION
**Pull request by karupayun**
We are moving all timer-related information to ClientData, to not have
any dependence between data and UI. We will avoid any include from data
to OrbitGl. As Blockchain is used for ScopeTree and has to be used in
ClientData but also is still being used in OrbitGl, we are moving both
classes to a new project named containers.

Bug we are focusing in: http://b/198135618

### CODE REVIEW

File: src/ClientData/include/ClientData/ScopeTree.h [hunk 1]
```diff
@@ -2,8 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#ifndef ORBIT_GL_SCOPE_TREE_H_
-#define ORBIT_GL_SCOPE_TREE_H_
+#ifndef CLIENT_DATA_SCOPE_TREE_H_
+#define CLIENT_DATA_SCOPE_TREE_H_
 
 #include <cstdint>
 #include <memory>
```
Comment (dimitry-) [inline src/ClientData/include/ClientData/ScopeTree.h:9]: also add a orbit_client_data namespace, here and for block chain

**Review by pierricgimmig — COMMENTED** [review]
I find it a bit strange that those low level classes are in ClientData. The blockchain data structure for example could be reused in other projects that wouldn't want a dependency on ClientData. 

BlockChain.h used to be in OrbitCore but was moved to up to OrbitGl because "It's only used in `OrbitGl`". BlockChain is now used both in ClientData and OrbitGl, but the dependency tree is not ideal. What about a "Containers" project that would hold BlockChain and ScopeTree?

**Comment by karupayun** [comment]
> I find it a bit strange that those low level classes are in ClientData. The blockchain data structure for example could be reused in other projects that wouldn't want a dependency on ClientData.
> 
> BlockChain.h used to be in OrbitCore but was moved to up to OrbitGl because "It's only used in `OrbitGl`". BlockChain is now used both in ClientData and OrbitGl, but the dependency tree is not ideal. What about a "Containers" project that would hold BlockChain and ScopeTree?

Yes, thanks Pierric, you might be right. I only was thinking on moving away of OrbitGl and it was a bit strange but I didn't find any better class. I will make a containers project.

File: src/ClientData/include/ClientData/ScopeTree.h (hunk 1 above)
Comment (karupayun) [reply src/ClientData/include/ClientData/ScopeTree.h]: Should I create a namespace orbit_containers?

Comment (dimitry-) [reply src/ClientData/include/ClientData/ScopeTree.h]: yes please.

File: src/OrbitGl/Batcher.h [hunk 2]
```diff
@@ -17,13 +17,14 @@
...
 
-#include "BlockChain.h"
+#include "Containers/BlockChain.h"
 #include "CoreMath.h"
 #include "Geometry.h"
 #include "PickingManager.h"
 #include "capture_data.pb.h"
 
 using TooltipCallback = std::function<std::string(PickingId)>;
+using orbit_containers::BlockChain;
```
Comment (dimitry-) [inline src/OrbitGl/Batcher.h:27]: please avoid `using namespace::class` in header files.

**Review by dimitry- — APPROVED** [review]

**Review by pierricgimmig — APPROVED** [review]
Thanks Pablo!

**Pull request merged**